src/Elastic.Apm/Model/Error.cs

```diff
@@ -55,19 +55,23 @@ private Error(Transaction transaction, string parentId, IApmLogger loggerArg, Di
 				}
 			}
 
-			CheckAndCaptureBaggage();
+			CheckAndCaptureBaggage(transaction);
 
 			IApmLogger logger = loggerArg?.Scoped($"{nameof(Error)}.{Id}");
 			logger.Trace()
 				?.Log("New Error instance created: {Error}. Time: {Time} (as timestamp: {Timestamp})",
 					this, TimeUtils.FormatTimestampForLog(Timestamp), Timestamp);
 		}
 
-		private void CheckAndCaptureBaggage()
+		private void CheckAndCaptureBaggage(Transaction transaction)
 		{
 			if (Activity.Current == null || !Activity.Current.Baggage.Any())
 				return;
 
+			//if context was not set prior we set it now to ensure we capture baggage for errors
+			//occuring during unsampled transactions
+			Context ??= transaction.Context.DeepCopy();
```

Comment:
Did the line below with `var labels = Context.InternalLabels.Value;` fail with a `NullReferenceException`? Or what was the behaviour before?

Reply:
Context is null because we never copy it from transaction when its an unsampled transaction.

Reply:
Thought about it but this way we only do it if there is actual baggage to set :)

Reply:
> Thought about it but this way we only do it if there is actual baggage to set :)

Yeah, that's a good point.